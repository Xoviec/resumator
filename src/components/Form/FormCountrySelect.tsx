import { Box, MenuItem } from "@mui/material";
import { VFC } from "react";
import { CountryMapping, getCountryIcon } from "../../lib";
import { FormSelect } from "../Form/FormSelect";

interface Props {
  name: string;
}

export const FormCountrySelect: VFC<Props> = ({ name }) => {
  return (
    <FormSelect
      size="small"
      required
      defaultValue={CountryMapping.NL.code}
      name={name}
      label="Country"
    >
      {Object.entries(CountryMapping).map(([key, info]) => {
        return (
          <MenuItem key={key} value={key}>
            <Box display="flex" flexDirection="row" gap="10px" alignItems="center">
              <Box display="flex" alignItems="center" style={{ width: "16px" }}>
                {getCountryIcon(info.code)}
              </Box>
              {info.title}
            </Box>
          </MenuItem>
        );
      })}
    </FormSelect>
  );
};
