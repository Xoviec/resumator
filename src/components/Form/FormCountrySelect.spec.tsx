import { fireEvent, render } from "@testing-library/react";
import { FunctionComponent } from "react";
import { FieldValues, FormProvider, useForm, UseFormReturn } from "react-hook-form";
import { FormCountrySelect } from "./FormCountrySelect";

describe("FormCountrySelect", () => {
  it("should render correctly", () => {
    const FormProviderWrapper: FunctionComponent = ({ children }) => {
      const form = useForm();

      return <FormProvider {...form}>{children}</FormProvider>;
    };

    const { asFragment } = render(
      <FormProviderWrapper>
        <FormCountrySelect name="countryCode" />
      </FormProviderWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should update field when value changes", () => {
    let form: UseFormReturn<FieldValues, Record<string, unknown>> = {} as any;
    const FormProviderWrapper: FunctionComponent = ({ children }) => {
      form = useForm();

      return <FormProvider {...form}>{children}</FormProvider>;
    };

    const { container } = render(
      <FormProviderWrapper>
        <FormCountrySelect name="countryCode" />
      </FormProviderWrapper>
    );

    const input = container.querySelector('input[name="countryCode"]');
    if (!input) return expect(false).toBeTruthy();
    fireEvent.change(input, { target: { value: "BE" } });

    expect(form.getValues()).toEqual({ countryCode: "BE" });
  });
});
