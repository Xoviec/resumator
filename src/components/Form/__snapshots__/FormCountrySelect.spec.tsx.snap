// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCountrySelect should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    required=""
    role="input"
    variant="outlined"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="mui-component-select-countryCode"
      class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="mui-component-select-countryCode"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiBox-root css-ks2x85"
      >
        <div
          class="MuiBox-root css-70qvj9"
          style="width: 16px;"
        >
          <svg
            id="flag-icons-nl"
            viewBox="0 0 512 512"
          >
            <path
              d="M0 0h512v512H0z"
              fill="#21468b"
            />
            <path
              d="M0 0h512v341.3H0z"
              fill="#fff"
            />
            <path
              d="M0 0h512v170.7H0z"
              fill="#ae1c28"
            />
          </svg>
        </div>
        The Netherlands
      </div>
    </div>
    <input
      aria-hidden="true"
      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      name="countryCode"
      required=""
      tabindex="-1"
      value="NL"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-hdw1oc"
      >
        <span
          class="notranslate"
        >
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</DocumentFragment>
`;
